src/sentry/incidents/model_temp.py

```diff
@@ -0,0 +1,999 @@
+                )
+            else:
+                raise Exception("Handler already registered for type %s" % type)
+            return handler
+
+        return inner
+
+    @classmethod
+    def get_registered_type(cls, type):
+        return cls._type_registrations[type]
+
+    @classmethod
+    def get_registered_types(cls):
+        return list(cls._type_registrations.values())
+
+
+class AlertRuleActivityType(Enum):
+    CREATED = 1
+    DELETED = 2
+    UPDATED = 3
+    ENABLED = 4
+    DISABLED = 5
+    SNAPSHOT = 6
+    ACTIVATED = 7
+    DEACTIVATED = 8
+
+
+@region_silo_only_model
+class AlertRuleActivity(Model):
+    """
+    Provides an audit log of activity for the alert rule
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+
+    alert_rule = FlexibleForeignKey("sentry.AlertRule")
+    previous_alert_rule = FlexibleForeignKey(
+        "sentry.AlertRule", null=True, related_name="previous_alert_rule"
+    )
+    user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete="SET_NULL")
+    type = models.IntegerField()
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_alertruleactivity"
+
+
+@register_alert_subscription_callback(AlertRuleMonitorType.ACTIVATED)
+def clean_expired_alerts(subscription: QuerySubscription) -> bool:
+    now = timezone.now()
+    subscription_end = subscription.date_added + timedelta(
+        seconds=subscription.snuba_query.time_window
+    )
+
+    if now > subscription_end:
+        delete_snuba_subscription(subscription)
+
+    return True
+
+
+post_delete.connect(AlertRuleManager.clear_subscription_cache, sender=QuerySubscription)
+post_save.connect(AlertRuleManager.clear_subscription_cache, sender=QuerySubscription)
+post_save.connect(AlertRuleManager.clear_alert_rule_subscription_caches, sender=AlertRule)
+post_delete.connect(AlertRuleManager.clear_alert_rule_subscription_caches, sender=AlertRule)
+
+post_delete.connect(AlertRuleTriggerManager.clear_alert_rule_trigger_cache, sender=AlertRule)
+post_save.connect(AlertRuleTriggerManager.clear_alert_rule_trigger_cache, sender=AlertRule)
+post_save.connect(AlertRuleTriggerManager.clear_trigger_cache, sender=AlertRuleTrigger)
+post_delete.connect(AlertRuleTriggerManager.clear_trigger_cache, sender=AlertRuleTrigger)
+
+post_save.connect(IncidentManager.clear_active_incident_cache, sender=Incident)
+post_save.connect(IncidentManager.clear_active_incident_project_cache, sender=IncidentProject)
+post_delete.connect(IncidentManager.clear_active_incident_project_cache, sender=IncidentProject)
+
+post_delete.connect(IncidentTriggerManager.clear_incident_cache, sender=Incident)
+post_save.connect(IncidentTriggerManager.clear_incident_trigger_cache, sender=IncidentTrigger)
+post_delete.connect(IncidentTriggerManager.clear_incident_trigger_cache, sender=IncidentTrigger)
```

Comment:
Most of the other change looks great!! I think maybe we might want to comment this portion out for now? I think it will actually register things and maybe we want to just leave that out now until we fully transition?